var/spack/repos/builtin/packages/ImageMagick/package.py

```diff
@@ -42,6 +43,9 @@ class Imagemagick(Package):
     depends_on('libtiff')
     depends_on('ghostscript')
 
+    def url_for_version(self, version):
+        return "https://github.com/ImageMagick/ImageMagick/archive/{0}.tar.gz".format(version)
```

Comment:
We can't use http://www.imagemagick.org/download/ because as @lee218llnl discovered, it changes at least monthly. GitHub is just a mirror for GitLab. We use mirrors of plenty of packages instead of the primary download repository.